src/learner.cc

```diff
@@ -300,13 +308,49 @@ class LearnerImpl : public Learner {
     for (auto& p_metric : metrics_) {
       p_metric->Configure({cfg_.begin(), cfg_.end()});
     }
+  }
 
+  void Load(dmlc::Stream* fi) override {
+    monitor_.Start(__func__);
+    if (cfg_.find("model_format") != cfg_.cend()) {
+      tparam_.model_format = cfg_.at("model_format");
+    }
+    if (tparam_.model_format == "json") {
+      LOG(WARNING) << "JSON serialization is at experimental stage. "
+                   << "Output schema is subject to change in the future.";
+      auto  ReadStream =
+          [](dmlc::Stream* fi) {
+            std::string buffer;
+            std::string temp;
+            size_t read {0};
+            size_t size {1};
```

Comment:
I would start with something bigger then 1 byte. I think 4K is a reasonable default